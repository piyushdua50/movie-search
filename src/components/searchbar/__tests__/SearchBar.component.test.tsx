import React from "react";
import { BrowserRouter, Routes, Route } from "react-router-dom";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import configureStore from "redux-mock-store";
import SearchBar from "../SearchBar.component";
test("render the SearchBar correctly", () => {
  const mockStore = configureStore();
  const { container } = render(
    <BrowserRouter>
      <Provider store={mockStore()}>
        <Routes>
          <Route path="/" element={<SearchBar />} />
        </Routes>
      </Provider>
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});
