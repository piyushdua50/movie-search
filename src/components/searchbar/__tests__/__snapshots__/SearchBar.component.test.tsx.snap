// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the SearchBar correctly 1`] = `
<div>
  <div
    class="searchContainer"
  >
    <div
      class="searchInputContainer"
    >
      <input
        class="searchInput"
        placeholder="Enter movie name to search......"
        type="text"
        value=""
      />
      <select
        class="searchInputButton"
      >
        <option
          value="ALL"
        >
          ALL
        </option>
        <option
          value="movie"
        >
          Movies
        </option>
        <option
          value="series"
        >
          Series
        </option>
        <option
          value="episode"
        >
          Episodes
        </option>
      </select>
    </div>
    <button
      class="searchButton"
    >
      Search
    </button>
  </div>
</div>
`;
